pilot/cmd/pilot-agent/options/options.go

```diff
@@ -119,4 +119,9 @@ var (
 	istiodSAN = env.RegisterStringVar("ISTIOD_SAN", "",
 		"Override the ServerName used to validate Istiod certificate. "+
 			"Can be used as an alternative to setting /etc/hosts for VMs - discovery address will be an IP:port")
+
+	minimumDrainDurationEnv = env.RegisterDurationVar("MINIMUM_DRAIN_DURATION",
+		5*time.Second,
```

Comment:
May be covered elsewhere, why is this an env var at all?
Is it controlling the “wait for connection termination “ behaviour or just when it starts polling for connection count?
If this time is 0 then we will start checking for connections right away which does not sound too bad either. Do we really need this var?

Reply:
> or just when it starts polling for connection count?

This is when It starts polling for connection count. This is to avoid race condition described here https://github.com/envoyproxy/envoy/issues/17056#issuecomment-886267587

Reply:
If we are worried about that race then we should not let users set this env var to 0. at what time do we deem that the race will not be there? I am still wondering about keeping this configurable. Again in remote istiod this is not configurable so it is important to just set the right value.

Reply:
Proxy env vars are configurable regardless of where istiod runs

Reply:
Ok my bad, I was reading this as a pilot env var.

How did we arrive at 5, and what process should a user go thru to decide on another value ?

Reply:
> How did we arrive at 5, and what process should a user go thru to decide on another value ?

5s was chosen just as we felt it is reasonable. We do not expect this to be tuned by users. The reason why we provided env var is so that users can set it to 0  - to get back the old behaviour.

Reply:
> 5s was chosen just as we felt it is reasonable. We do not expect this to be tuned by users. The reason why we provided env var is so that users can set it to 0 - to get back the old behaviour.

The comment for this env should reflect the user can change it to 0 if they wish to get back the old behavior.

Reply:
@howardjohn with the change to the new approach do you think this is still needed or we can just internally hard code it to `5s`?